server/src/main/java/org/apache/iotdb/db/metrics/server/ServerArgument.java

```diff
@@ -0,0 +1,347 @@
+		return totalPhysicalMemory;
+	}
+
+	public long getFreePhysicalMemory() {
+		return freePhysicalMemory;
+	}
+
+	public long getUsedPhysicalMemory() {
+		return usedPhysicalMemory;
+	}
+
+	public double getCpuRatio() {
+		String osName = System.getProperty("os.name").toLowerCase();
+		cpuRatio = 0;
+		if (osName.indexOf("windows") >= 0) {
+			cpuRatio = getCpuRatioForWindows();
+		} else if(osName.indexOf("linux") >= 0){
+			cpuRatio = getCpuRateForLinux();
+		} else {
+			try {
+				throw new Exception("Can't get the cpu ratio,because this OS is not support");
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		return cpuRatio;
+	}
+
+	/**
+	 * cpu ratio for linux
+	 */
+	private double getCpuRateForLinux() {
+		try {
+			long[] c0 = readLinuxCpu();
+			Thread.sleep(CPUTIME);
+			long[] c1 = readLinuxCpu();
+			if (c0 != null && c1 != null) {
+				long idleCpuTime = c1[0] - c0[0];
+				long totalCpuTime = c1[1] - c0[1];
+				if (totalCpuTime == 0) {
+					return 100.0;
+				}
+				return Double.valueOf(100 * (1 - idleCpuTime / totalCpuTime)).doubleValue();
+			} else {
+				return 0.0;
+			}
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+			return 0.0;
+		}
+	}
+
+	/**
+	 * cpu ratio for windows
+	 */
+	private double getCpuRatioForWindows() {
+		try {
+			String procCmd = System.getenv("windir") + "\\system32\\wbem\\wmic.exe process get Caption,CommandLine,"
+					+ "KernelModeTime,ReadOperationCount,ThreadCount,UserModeTime,WriteOperationCount";
+			long[] c0 = readWinCpu(Runtime.getRuntime().exec(procCmd));
+			Thread.sleep(CPUTIME);
+			long[] c1 = readWinCpu(Runtime.getRuntime().exec(procCmd));
+			if (c0 != null && c1 != null) {
+				long idletime = c1[0] - c0[0];
+				long busytime = c1[1] - c0[1];
+				if ((busytime + idletime) == 0) {
+					return 100.0;
+				}
+				return Double.valueOf(100 * (busytime) / (busytime + idletime)).doubleValue();
+			} else {
+				return 0.0;
+			}
+		} catch (Exception ex) {
+			ex.printStackTrace();
+			return 0.0;
+		}
+	}
+
+	/**
+	 * read cpu info(windows)
+	 */
+	private long[] readWinCpu(final Process proc) {
+		long[] retn = new long[2];
+		try {
+			proc.getOutputStream().close();
+			InputStreamReader ir = new InputStreamReader(proc.getInputStream());
+			LineNumberReader input = new LineNumberReader(ir);
+			String line = input.readLine();
+			if (line == null || line.length() < 10) {
+				return null;
+			}
+			int capidx = line.indexOf("Caption");
+			int cmdidx = line.indexOf("CommandLine");
+			int rocidx = line.indexOf("ReadOperationCount");
+			int umtidx = line.indexOf("UserModeTime");
+			int kmtidx = line.indexOf("KernelModeTime");
+			int wocidx = line.indexOf("WriteOperationCount");
+			long idletime = 0;
+			long kneltime = 0;
+			long usertime = 0;
+			while ((line = input.readLine()) != null) {
+				if (line.length() < wocidx) {
+					continue;
+				}
+				String caption = substring(line, capidx, cmdidx - 1).trim();

```

Comment:
I think you can simply use Java API to get substring, and notice the range of `i`. **Other places in this PR** are the same.
```suggestion
				String caption = line.substring(capidx, cmdidx).trim();
```